pa-rserve/src/main/java/org/ow2/parserve/util/rsession/StartRserve.java

```diff
@@ -0,0 +1,253 @@
+package org.ow2.parserve.util.rsession;
+
+import org.apache.log4j.Logger;
+import org.rosuda.REngine.Rserve.RConnection;
+
+import java.io.File;
+
+/**
+ * simple class that start Rserve locally if it's not running already - see mainly <code>checkLocalRserve</code> method. It spits out quite some debugging outout of the console, so feel free to modify it for your application if desired.<p>
+ * <i>Important:</i> All applications should shutdown every Rserve that they started! Never leave Rserve running if you started it after your application quits since it may pose a security risk. Inform the user if you started an Rserve instance.
+ */
+public class StartRserve {
+
+    private static final Logger logger = Logger.getLogger(StartRserve.class);
+    public static String DEFAULT_REPOS = "http://cran.irsn.fr/";
+    public static Process rProcess;
+
+    /**
+     * R batch to check Rserve is installed
+     *
+     * @param Rcmd command necessary to start R
+     * @return Rserve is already installed
+     */
+    public static boolean isRserveInstalled(String Rcmd) {
+        StringBuffer result = new StringBuffer();
+        boolean done = doInR("i=installed.packages();is.element(set=i,el='Rserve')", Rcmd, "--vanilla -q", true, result, result);
+        if (!done) {
+            return false;
+        }
+        //System.err.println("output=\n===========\n" + result.toString() + "\n===========\n");
+        if (result.toString().contains("TRUE")) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * R batch to install Rserve
+     *
+     * @param Rcmd       command necessary to start R
+     * @param http_proxy http://login:password@proxy:port string to enable internet access to rforge server
+     * @return success
+     */
+    public static boolean installRserve(String Rcmd, String http_proxy, String repository) {
+        if (repository == null || repository.length() == 0) {
+            repository = DEFAULT_REPOS;
+        }
+        logger.info("Install Rserve from " + repository + " ... (http_proxy=" + http_proxy + ") ");
+        boolean ok = doInR((http_proxy != null ? "Sys.setenv(http_proxy=" + http_proxy + ");" : "") + "install.packages('Rserve',repos='" + repository + "')", Rcmd, "--vanilla", true, null, null);
+        if (!ok) {
+            logger.error("RServe installation failed");
+            return false;
+        }
+        int n = 5;
+        while (n > 0) {
+            try {
+                Thread.sleep(10000 / n);
```

Comment:
Is it not possible to get return code of the R process or at least to parse output?